lib/create-eslint-config/config.ts

```diff
@@ -0,0 +1,398 @@
+
+import pluginAva from 'eslint-plugin-ava';
+import pluginUnicorn from 'eslint-plugin-unicorn';
+import pluginImport from 'eslint-plugin-import-x';
+import pluginN from 'eslint-plugin-n';
+import pluginComments from '@eslint-community/eslint-plugin-eslint-comments';
+import pluginPromise from 'eslint-plugin-promise';
+import pluginNoUseExtendNative from 'eslint-plugin-no-use-extend-native';
+import configXoTypescript from 'eslint-config-xo-typescript';
+import stylisticPlugin from '@stylistic/eslint-plugin';
+import globals from 'globals';
+import {type Linter} from 'eslint';
+import {
+  DEFAULT_IGNORES,
+  TS_EXTENSIONS,
+  TS_FILES_GLOB,
+  ALL_FILES_GLOB,
+  JS_EXTENSIONS,
+  ALL_EXTENSIONS,
+} from '../constants.js';
+
+if (Array.isArray(pluginAva?.configs?.['recommended'])) {
+  throw new TypeError('Invalid eslint-plugin-ava');
+}
+
+if (!configXoTypescript[1]) {
+  throw new Error('Invalid eslint-config-xo-typescript');
+}
+
+/**
+ * The base config that xo builds on top of from user options
+ */
+export const config: Linter.Config[] = [
+  {
+    ignores: DEFAULT_IGNORES,
+  },
+  {
+    name: 'XO',
+    files: [ALL_FILES_GLOB],
+    plugins: {
+      'no-use-extend-native': pluginNoUseExtendNative,
+      ava: pluginAva,
+      unicorn: pluginUnicorn,
+      'import-x': pluginImport,
+      n: pluginN,
+      '@eslint-community/eslint-comments': pluginComments,
+      promise: pluginPromise,
+      // @ts-expect-error: This is a private plugin
+      '@stylistic': stylisticPlugin,
+    },
+    languageOptions: {
+      globals: {
+        ...globals.es2021,
+        ...globals.node,
+      },
+      ecmaVersion: configXoTypescript[0]?.languageOptions?.ecmaVersion,
+      sourceType: configXoTypescript[0]?.languageOptions?.sourceType,
+      parserOptions: {
+        ...configXoTypescript[0]?.languageOptions?.parserOptions,
+      },
+    },
+    settings: {
+      'import-x/extensions': ALL_EXTENSIONS,
+      'import-x/core-modules': ['electron', 'atom'],
+      'import-x/parsers': {
+        espree: JS_EXTENSIONS,
+        '@typescript-eslint/parser': TS_EXTENSIONS,
+      },
+      'import-x/external-module-folders': [
+        'node_modules',
+        'node_modules/@types',
+      ],
+      'import-x/resolver': {
+        node: ALL_EXTENSIONS,
+      },
+    },
+    /**
+     * These are the base rules that are always applied to all js and ts file types
+     */
+    rules: {
+      ...pluginAva?.configs?.['recommended']?.rules,
+      ...pluginUnicorn.configs?.recommended?.rules,
+      'no-use-extend-native/no-use-extend-native': 'error',
+      // TODO: Remove this override at some point.
+      // It's just here to ease users into readable variable names.
+      'unicorn/prevent-abbreviations': [
```

Comment:
Wait until someone else confirms, but I think these are complete duplicates and can now be dropped since they appear in unicorn itself. 

The other plugins should also follow once their config is fully moved to eslint-config-xo, but that can be a follow-up (unless they're already there too)

Reply:
Open an issue instead. It's not really related to this PR. This PR is already huge.